Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/ORTHelpers.cpp

```diff
@@ -0,0 +1,157 @@
+    OrtSessionOptionsAppendExecutionProvider_DML(sessionOptions, 0);
+
+    return Ort::Session(ortEnvironment, model_file_path, sessionOptions);
+}
+
+
+// Run the buffer through a preprocessing model that will shrink the
+// image from 512 x 512 x 4 to 224 x 224 x 3
+Ort::Value Preprocess(Ort::Session& session,
+    ComPtr<ID3D12Resource> currentBuffer)
+{
+    // Init OrtAPI
+    OrtApi const& ortApi = Ort::GetApi(); // Uses ORT_API_VERSION
+    const OrtDmlApi* ortDmlApi;
+    THROW_IF_NOT_OK(ortApi.GetExecutionProviderApi("DML", ORT_API_VERSION, reinterpret_cast<const void**>(&ortDmlApi)));
+
+    // Create ORT Value from buffer currently being drawn to screen
+    const char* memoryInformationName = "DML";
+    Ort::MemoryInfo memoryInformation(memoryInformationName, OrtAllocatorType::OrtDeviceAllocator, 0, OrtMemType::OrtMemTypeDefault);
+    ComPtr<IUnknown> inputTensorEpWrapper;
+    
+    // Calculate input shape
+    auto width = 800;
+    auto height = 600;
+    auto rowPitchInBytes = (width * 4 + 255) & ~255;
+    auto rowPitchInPixels = rowPitchInBytes / 4;
+    auto bufferInBytes = rowPitchInBytes * height;
+    const std::array<int64_t, 2> inputShape = { 1, bufferInBytes };
+
+    Ort::Value inputTensor = CreateTensorValueFromD3DResource(
+        *ortDmlApi,
+        memoryInformation,
+        currentBuffer.Get(),
+        inputShape,
+        ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT8,
+        /*out*/ IID_PPV_ARGS_Helper(inputTensorEpWrapper.GetAddressOf())
+    );
+
+    // Create input and output node names
+    const char* inputTensorName = "Input";
+    const char* outputTensorName = "Output";
+    std::vector<const char*> input_node_names;
+    input_node_names.push_back(inputTensorName);
+    std::vector<const char*> output_node_names;
+    output_node_names.push_back(outputTensorName);
+    
+    // Evaluate input (resize from 512 x 512 x 4 to 224 x 224 x 3)
+    Ort::Value outputTensor(nullptr);
+    session.Run(Ort::RunOptions{ nullptr }, input_node_names.data(),
+        &inputTensor, 1, output_node_names.data(), &outputTensor, 1);
+
+    return outputTensor;
+}
+
+// Classify the image using EfficientNet and return the results
+winrt::com_array<float> Eval(Ort::Session& session, const Ort::Value& prev_input) {
+    // Create input and output node names
+    const char* inputTensorName = "images:0";
+    const char* outputTensorName = "Softmax:0";
+    std::vector<const char*> input_node_names;
+    input_node_names.push_back(inputTensorName);
+    std::vector<const char*> output_node_names;
+    output_node_names.push_back(outputTensorName);
+
+    // Evaluate
+    auto output_tensors = session.Run(Ort::RunOptions{ nullptr }, input_node_names.data(), &prev_input, 1, output_node_names.data(), 1);
+    
+    // Get the 1000 EfficientNet classifications as a com_array and return
+    // the results
+    float* floatarr = output_tensors.front().GetTensorMutableData<float>();
```

Comment:
>floatarr [](http://example.com/codeflow?start=11&length=8)

[nit] With very rare except of really common cases (e.g. ptr), we should spell out whole words.

`floatArray`

Reply:
Done.